resources/assets/app.css

```diff
@@ -17,4 +17,51 @@
 @tailwind components;
 @tailwind utilities;
 
-[x-cloak] { display: none !important; }
+.table-of-contents {
+    @apply pb-3;
+}
+
+.table-of-contents > li {
+    @apply my-[0.35rem];
+}
+
+.table-of-contents ul {
+    @apply pl-2;
+}
+
+.table-of-contents a {
+    @apply block -ml-8 pl-8 opacity-80 hover:opacity-100 hover:bg-gray-200/20 transition-all duration-300 relative;
+}
+
+.table-of-contents a::before {
+    @apply content-['#'] text-[75%] opacity-50 mr-1 transition-opacity duration-300;
+}
+
+.table-of-contents a:hover::before {
+    @apply opacity-100;
+}
+
+/**
+ * Heading Permalinks
+ */
+
+.prose h1, .prose h2, .prose h3, .prose h4, .prose h5, .prose h6 {
+    @apply w-fit;
+}
+
+.prose :is(h1,h2,h3,h4,h5,h6):hover .heading-permalink,
+.prose :is(h1,h2,h3,h4,h5,h6):focus .heading-permalink {
+    @apply opacity-75 grayscale transition-opacity duration-100 ease-out;
+}
+
+.heading-permalink {
+    @apply opacity-0 ml-1 transition-opacity duration-300 ease-linear px-1 scroll-m-4;
+}
+
+.heading-permalink::before {
+    @apply content-['#'];
+}
+
+.heading-permalink:hover, .heading-permalink:focus {
+    @apply opacity-100 grayscale-0;
+}
```

Comment:
Also, ideally if we could have a post-processor (or custom CommonMark heading renderer?) to swap out headings to a customizable Blade component to fully support changing all parts of heading permalinks from Blade would be great, for example moving them around and stuff would be really nice. Also supporting changing the icons and stuff too of course.

Reply:
Now tracked in https://github.com/hydephp/develop/pull/2006

Reply:
Now handled in https://github.com/hydephp/develop/pull/2047